x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -650,8 +651,28 @@ static boolean match(Task task, String expectedId) {
     }
 
     public static long estimateMemoryUsageBytes(String modelId, long totalDefinitionLength) {
-        // While loading the model in the process we need twice the model size.
-        return isElserModel(modelId) ? ELSER_1_MEMORY_USAGE.getBytes() : MEMORY_OVERHEAD.getBytes() + 2 * totalDefinitionLength;
+        return estimateMemoryUsageBytes(modelId, totalDefinitionLength);
```

Comment:
Should this call the other version of the method? This looks like a recursive call as it stands.